inline_hook.cpp

```diff
@@ -84,6 +87,17 @@ void GenerateJmpLarge(char* pWriteTo, void* pTarget)
 	pWriteTo += 4;
 	*pWriteTo = 0xc3;
 }
+/*
+NOTE: use long far jmp
+jmp addr64
+ff 25 00 00 00 00 addr
+ */
+void GenerateJmpLarge(char* pWriteTo, void* pTarget)

```

Comment:
This assembly seem not correct. I tried the bytes `ff 25 00 00 00 00, 01 02 03 04 05 06 07 08`, which should be `jmp 0x0102030405060708', but it turns out to be:

menooker@DESKTOP:/tmp$ echo 'asm(".byte 0xff, 0x25, 0x00,0x00,0x00,0x00,0x01,0x02,0x03,0x04,0x05,0x06,0x07,0x08");' > test.c; gcc -c test.c ;objdump -d test.o

test.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <.text>:
   0:   ff 25 00 00 00 00       jmpq   *0x0(%rip)        # 0x6
   6:   01 02                   add    %eax,(%rdx)
   8:   03                      .byte 0x3
   9:   04 05                   add    $0x5,%al
   b:   06                      (bad)
   c:   07                      (bad)
   d:   08                      .byte 0x8

There should be no jmp absolute in x64. See reply `There is no jump of the form JMP absaddr to an absolute address in 64 bit mode. ` in [StackOverflow](https://stackoverflow.com/a/9819716)

Reply:
jmpq QWORD PTR[rip];
.QWORD newfunc address

Reply:
checked